### PR TITLE
fix: flox -> Flox

### PR DESCRIPTION
**Pull request by rossturk**
## Proposed Changes

Replace "flox" with "Flox" in user-visible locations. Most importantly, update the metrics notification that users see upon first invocation.

## Release Notes

N/A

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -19,6 +19,6 @@ Expected:
 
 **Screenshots/terminal output (if relevant):**
 
-**flox Version (run `flox --version` if possible):**
+**Flox Version (run `flox --version` if possible):**
```
Comment (mkenigs) [inline .github/ISSUE_TEMPLATE/bug-report.md:22]: nonblocking: in this case it's referring the version of the CLI, so my understanding was that we leave that lowercase

File: cli/flox/src/utils/init/metrics.rs [hunk 2]
```diff
@@ -72,9 +72,9 @@ pub fn init_telemetry_uuid(data_dir: impl AsRef<Path>, cache_dir: impl AsRef<Pat
     debug!("Created new telemetry UUID: {}", telemetry_uuid);
 
     let notice = formatdoc! {"
-        flox collects basic usage metrics in order to improve the user experience.
+        Flox collects basic usage metrics in order to improve the user experience.
```
Comment (mkenigs) [inline cli/flox/src/utils/init/metrics.rs:75]: nonblocking: same here

File: cli/flox/src/commands/mod.rs [hunk 3]
```diff
@@ -91,9 +91,9 @@ const UPDATE_NOTIFICATION_FILE_NAME: &str = "update-notification.json";
 const UPDATE_NOTIFICATION_EXPIRY: Duration = Duration::days(1);
 
 static FLOX_DESCRIPTION: &'_ str = indoc! {"
-    flox is a virtual environment and package manager all in one.\n\n
+    Flox is a virtual environment and package manager all in one.\n\n
```
Comment (mkenigs) [inline cli/flox/src/commands/mod.rs:94]: nonblocking: probably same here?

**Review by mkenigs — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (rossturk) [reply .github/ISSUE_TEMPLATE/bug-report.md:22]: The way I interpret the style guide, it is `flox` when referring to subcommands but `Flox` at all other times. So we might say "you should use the flox activate command" but would not say "flox Version: [version]".

File: cli/flox/src/utils/init/metrics.rs (hunk 2 above)
Comment (rossturk) [reply cli/flox/src/utils/init/metrics.rs:75]: In this case, I believe we are referring to Flox as an entity - i.e. we collect basic usage metrics. But if I interpret it as "the Flox CLI collects metrics" I still think it's uppercase as it's not used in the context of a subcommand.

**Pull request merged**